inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp

```diff
@@ -150,6 +151,9 @@ bool ngraph::pass::CommonOptimizations::run_on_function(std::shared_ptr<ngraph::
     // CF is required after all decompositions
     manager.register_pass<ngraph::pass::ConstantFolding>();
 
+    // TransposeReshapeEliminationForMatmul must be after EinsumDecomposition
+    manager.register_pass<ngraph::pass::TransposeReshapeEliminationForMatmul>();
```

Comment:
You can put this pass together with other passes in decomp. So no additional graph traversals will be performed.